packages/core/src/components/fields/LayoutGridField.tsx

```diff
@@ -363,612 +947,36 @@ function isNumericIndex(str: string) {
-          name={name}
-          required={isRequired}
-          readonly={uiReadonly}
-          schema={schema}
-          uiSchema={fieldUiSchema}
-          errorSchema={get(errorSchema, name)}
-          fieldPathId={fieldIdSchema}
-          formData={get(formData, name)}
-          onChange={this.onFieldChange(name)}
-          onBlur={onBlur}
-          onFocus={onFocus}
-          options={optionsInfo?.options}
-          registry={registry}
-        />
-      );
-    }
-
-    if (UIComponent) {
-      return (
-        <UIComponent
-          data-testid={LayoutGridField.TEST_IDS.uiComponent}
-          {...otherProps}
-          name={name}
-          required={isRequired}
-          formData={formData}
-          readOnly={!!isReadonly || readonly}
-          errorSchema={errorSchema}
-          uiSchema={uiSchema}
-          schema={initialSchema}
-          fieldPathId={fieldPathId}
-          onBlur={onBlur}
-          onFocus={onFocus}
-          registry={registry}
-          {...uiProps}
-        />
-      );
+    if (GridType.COLUMN in layoutGridSchema) {
+      return <LayoutGridCol {...props} layoutGridSchema={layoutGridSchema} />;
     }
-    return null;
-  }
-
-  /** Renders the `LayoutGridField`. If there isn't a `layoutGridSchema` prop defined, then try pulling it out of the
-   * `uiSchema` via `ui:LayoutGridField`. If `layoutGridSchema` is an object, then check to see if any of the properties
-   * match one of the `GridType`s. If so, call the appropriate render function for the type. Otherwise, just call the
-   * generic `renderField()` function with the `layoutGridSchema`.
-   *
-   * @returns - the rendered `LayoutGridField`
-   */
-  render() {
-    const { uiSchema } = this.props;
-    let { layoutGridSchema } = this.props;
-    const uiOptions = getUiOptions<T, S, F>(uiSchema);
-    if (!layoutGridSchema && LAYOUT_GRID_UI_OPTION in uiOptions && isObject(uiOptions[LAYOUT_GRID_UI_OPTION])) {
-      layoutGridSchema = uiOptions[LAYOUT_GRID_UI_OPTION];
+    if (GridType.COLUMNS in layoutGridSchema) {
+      return <LayoutGridColumns {...props} layoutGridSchema={layoutGridSchema} />;
     }
-
-    if (isObject(layoutGridSchema)) {
-      if (GridType.ROW in layoutGridSchema) {
-        return this.renderRow(layoutGridSchema as GridSchemaType);
-      }
-      if (GridType.COLUMN in layoutGridSchema) {
-        return this.renderCol(layoutGridSchema as GridSchemaType);
-      }
-      if (GridType.COLUMNS in layoutGridSchema) {
-        return this.renderColumns(layoutGridSchema as GridSchemaType);
-      }
-      if (GridType.CONDITION in layoutGridSchema) {
-        return this.renderCondition(layoutGridSchema as GridSchemaType);
-      }
+    if (GridType.CONDITION in layoutGridSchema) {
+      return <LayoutGridCondition {...props} layoutGridSchema={layoutGridSchema} />;
     }
-    return this.renderField(layoutGridSchema);
   }
+  return <LayoutGridFieldComponent {...props} gridSchema={layoutGridSchema} />;
 }
```

Comment:
Since my company relies on the `LayoutGridField.TEST_IDS` variable, can you add the following? Plus that will simplify the tests since you can revert a bunch of changes:
```suggestion
}

LayoutGridField.TEST_IDS = LAYOUT_GRID_FIELD_TEST_IDS;

```